### PR TITLE
Add support for BlockBundle 4.x

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

@core23 
 After check #488 I think we can add support for BlockBundle 4.x. Can u check it?

I am targeting this branch, because it is PATCH.

### CODE REVIEW

**Comment by wbloszyk** [comment]
@core23 What do you think about it? Can we remove AbstractAdminBlockService? 
It is not used in any place. You can find only method call like here:
https://github.com/sonata-project/SonataPageBundle/blob/master/src/Admin/BlockAdmin.php#L217-L238

@sonata-project/contribiutors
If we can do it like BC, then major upgrade for other sonata-project bundles will be much simpler.

File: src/Block/Service/AbstractClassificationBlockService.php [hunk 1]
```diff
@@ -56,6 +69,98 @@ public function __construct(
...
+        $this->buildEditForm($formMapper, $block);
+    }
+
+    /**
+     * NEXT_MAJOR: Remove this method.
+     *
+     * @deprecated since sonata-project/block-bundle 3.16, to be removed in version 4.0.
+     */
+    public function buildEditForm(FormMapper $form, BlockInterface $block)
+    {
```
Comment (core23) [inline src/Block/Service/AbstractClassificationBlockService.php]:
```suggestion
    {
         $this->buildEditForm($formMapper, $block);
```

**Comment by core23** [comment]
Looking at Travis, sonata-block 4 is never resolved

File: src/Block/Service/AbstractClassificationBlockService.php (hunk 1 above)
Comment (wbloszyk) [reply src/Block/Service/AbstractClassificationBlockService.php]: This is AbstractAdminBlockService method, we should not change then. Self reference method make infinity loop.

**Comment by wbloszyk** [comment]
@core23 
I think other dependencies should be update first (like MediaBundle). What do you think about this changes? Should I work on this?

**Comment by core23** [comment]
> @core23
> I think other dependencies should be update first (like MediaBundle). What do you think about this changes? Should I work on this?

As said several time on many different issues. We had some major changes on the old core bundle and the block bundle. This changes can only be resolved with new major versions on our projects. 

The main goal is still a new admin bundle major version. After that we can try to find a way for all other projects.

File: composer.json [hunk 2]
```diff
@@ -23,8 +23,8 @@
     "require": {
         "php": "^7.1",
         "cocur/slugify": "^2.0 || ^3.0 || ^4.0",
-        "sonata-project/admin-bundle": "^3.31",
-        "sonata-project/core-bundle": "^3.14",
+        "sonata-project/admin-bundle": "^3.31 || dev-master",
```
Comment (core23) [inline composer.json:26]: We should not add this to the stable branch

**Review by core23 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**